flixel/system/render/hardware/gl/GLRenderHelper.hx

```diff
@@ -0,0 +1,352 @@
+		
+		createBuffer();
+	}
+	
+	private function createBuffer():Void
+	{
+		if (_buffer != null)
+			GL.deleteBuffer(_buffer);
+		
+		var uv = _texture.renderTexture.uvData;
+		
+		var vertices:Float32Array = new Float32Array([
+			-1.0, 	-1.0, 	uv.x, 		uv.y,
+			1.0, 	-1.0, 	uv.width, 	uv.y,
+			-1.0, 	1.0, 	uv.x, 		uv.height,
+			1.0, 	-1.0, 	uv.width, 	uv.y,
+			1.0, 	1.0, 	uv.width, 	uv.height,
+			-1.0, 	1.0, 	uv.x, 		uv.height
+		]);
+		
+		_buffer = GL.createBuffer();
+		GL.bindBuffer(GL.ARRAY_BUFFER, _buffer);
+		GL.bufferData(GL.ARRAY_BUFFER, vertices, GL.STATIC_DRAW);
+		GL.bindBuffer(GL.ARRAY_BUFFER, null);
+	}
+	
+	public function destroy():Void
+	{
+		_texture = FlxDestroyUtil.destroy(_texture);
+		
+		object = null;
+		_objMatrixCopy = null;
+		_renderMatrix = null;
+		_scissor = null;
+		_viewport = null;
+		
+		if (_buffer != null)
+			GL.deleteBuffer(_buffer);
+	}
+	
+	private function get_numPasses():Int
+	{
+		return getObjectNumPasses(object);
+	}
+	
+	/**
+	 * Start capturing graphics to internal buffer
+	 * 
+	 * @param	fullscreen			Do we need to grab whole screen area?
+	 */
+	public function capture(fullscreen:Bool = false):Void
+	{
+		if (numPasses <= 0)
+			return;
+		
+		_scissorDisabled = (GL.getParameter(GL.SCISSOR_TEST) == 1) && !fullscreen;
+		
+		if (_scissorDisabled)
+		{
+			_scissor = GL.getParameter(GL.SCISSOR_BOX);
+			
+			// we need to disable scissor testing while we are drawing effects
+			GL.disable(GL.SCISSOR_TEST);
+			GL.scissor(0, 0, 0, 0);
+		}
+		
+		var objectTransfrom:Matrix = object.__worldTransform;
+		_objMatrixCopy.copyFrom(objectTransfrom);
+		_fullscreen = fullscreen;
+		
+		if (!fullscreen)
+		{
+			// we need to resize viewport and store its value
+			_viewport = GL.getParameter(GL.VIEWPORT);
+			GL.viewport(0, 0, width, height);
+			
+			// and let's reset object matrix
+			objectTransfrom.identity();
+		}
+		
+		FrameBufferManager.push(_texture.renderTexture.frameBuffer);
+		_texture.clear(0, 0, 0, 1.0, GL.DEPTH_BUFFER_BIT | GL.COLOR_BUFFER_BIT);
+	}
+	
+	/**
+	 * Actual rendering of effects added to display object through FlxShaderFilters
+	 * 
+	 * @param	renderSession	current render session (see openfl internals for details)
+	 */
+	public function render(renderSession:RenderSession):Void
+	{
+		var passes:Int = numPasses;
+		if (passes <= 0)	return;
+		
+		var gl:GLRenderContext = renderSession.gl;
+		var renderer:GLRenderer = cast renderSession.renderer;
+		var textureToUse:GLTexture;
+		
+		var objectTransfrom:Matrix = object.__worldTransform;
+		
+		var filters:Array<BitmapFilter> = object.filters;
+		var numFilters:Int = filters.length;
+		var shader:FlxShader = null;
+		
+		var filterIndex:Int = 0;
+		var i:Int = 0;
+		
+		while (i < passes && filterIndex < numFilters)
+		{
+			shader = null;
+			
+			if (Std.is(filters[filterIndex], ShaderFilter))
+			{
+				shader = cast(filters[filterIndex], ShaderFilter).shader;
```

Comment:
Probably shouldn't use a safe cast here - they have unnecessary runtime overhead (unnecessary you're already checking with `Std.is()` anyway - safe casts are only necessary when you want to try-catch it).

https://haxe.org/manual/expression-cast.html